### ISSUE TITLE
New format proposal

### ISSUE THREAD

**Issue by nobu-k** [sensorbee/opencv#1]
```
{
  "format": "cvmat",
  "width": width,
  "height": height,
  "image": <blob>
}
```

The current tuples generated from video sources look ilke this. In raspivideo, it also wanted to output raw images and JPEGs. So, I defined the format like below:

```
{
  "format": "raw",
  "color_model": "rgb",
  "width": width,
  "height": height,
  "image": <blob>
}
```

Instead of "cvmat", I defined "raw" format containing raw pixel values in a layout specified by the "color_model" field. The "color_model" field is only provided when the format is "raw". "raw" + "bgr" is equivalent to the current "cvmat" format.

I think this is more general and flexible than using the term "cvmat". What do you think?

**Comment by disktnk** [comment]
> The "color_model" field is only provided when the format is "raw".

sounds great.

If an image format is "RAW" and "RGB**A**", then `format` is "raw" and `color_model` is "rgba" ? I agree when `color_model`represents `image` binary style.

**Comment by nobu-k** [comment]
Yes. Even though not all plugins need to support wide variety of color models, having that field would be useful.
